### ISSUE TITLE
[Bug]: jsx-handler-names reports error with inline ternary conditional callbacks

### ISSUE THREAD

**Issue by stevensacks** [jsx-eslint/eslint-plugin-react#3855]
### Is there an existing issue for this?

- [x] I have searched the existing issues and my issue is unique
- [x] My issue appears in the command-line and not only in the text editor

### Description Overview

Conditionally applying callback functions with inline ternary statements violates the rule, even when the condition will satisfy the rule.

```tsx
<MyComponent
  onClickNext={isNextDisabled ? undefined : handleNext}
/>
```

```
ESLint: Handler function for onClickNext prop key must be a camelCase name beginning with 'handle' only(react/ jsx-handler-names)
```

### Expected Behavior

If there is an inline ternary condition and it returns the properly named function, it should pass.

### eslint-plugin-react version

v7.37.2

### eslint version

v8.57.0

### node version

v20.17.0

**Comment by stevensacks** [comment]
I hope it's ok that I've changed the name to Enhancement. Since it's not technically a bug because it is not written to support this, it's a request for enhancement.
